samples/openapi3/client/petstore/typescript/builds/composed-schemas/models/PetsPatchRequest.ts

```diff
@@ -14,49 +14,11 @@ import { Cat } from '../models/Cat';
 import { Dog } from '../models/Dog';
 import { HttpFile } from '../http/http';
 
-export class PetsPatchRequest {
-    'hunts'?: boolean;
-    'age'?: number;
-    'bark'?: boolean;
-    'breed'?: PetsPatchRequestBreedEnum;
-
-    static readonly discriminator: string | undefined = "petType";
```

Comment:
this changes the discriminator value to undefined, since we have

https://github.com/OpenAPITools/openapi-generator/blob/8473d47982073e99f74c556ac229465c827b7a6a/samples/openapi3/client/petstore/typescript/builds/composed-schemas/models/Dog.ts#L19

is this intentional?

Reply:
Yes, it is intentional change
There is oneOf in the request, now it works correctly
https://github.com/OpenAPITools/openapi-generator/blob/011acc99509af37e349425d632d75a775c15f579/modules/openapi-generator/src/test/resources/3_0/composed-schemas.yaml#L17-L20

Reply:
Yes, intentional
I've created type and class:
```typescript
/**
 * @type PetsPatchRequest
 * Type
 * @export
 */
export type PetsPatchRequest = Cat | Dog;

/**
* @type PetsPatchRequestClass
* @export
*/
export class PetsPatchRequestClass {
        static readonly discriminator: string | undefined = "petType";
}

```

https://github.com/OpenAPITools/openapi-generator/pull/19027/files#diff-531a756a09a9bb02d3916047a7b0c39dc8caf49c638c9c74391ad432166cd579R25

Reply:
Yes, it is 
Should I make a PR to 8.0.x?

Reply:
so you are saying the currently generated classes (before this PR) for the typescript generator dont work? what does not work?

Reply:
I've researched it one more time and I'm sorry for incorrect information.
It works in some cases, but not every time (depending from swagger file)
This doesn't be backwards compatible.

Reply:
https://github.com/OpenAPITools/openapi-generator/pull/19252

Reply:
just so i correctly understand, can you update the PR description with the motivation why this change is needed, what use case it solves, and why there are two types generated (the union and the class), and why the class does not have any fields besides a mapping and discriminator?

Reply:
Any updates?
@macjohnny

Reply:
Any updates?
@macjohnny  @wing328

Reply:
Will check soon. Sorry, for the dela